addons/website_event_track_quiz/controllers/main.py

```diff
@@ -0,0 +1,144 @@
+import json
+
+from odoo import http
+from odoo.addons.website_event_track_session.controllers.session import WebsiteEventSessionController
+from odoo.exceptions import AccessError, UserError
+from odoo.http import request
+
+
+class WebsiteEventTrackQuiz(WebsiteEventSessionController):
+    def _fetch_track(self, track_id):
+        track = request.env['event.track'].browse(int(track_id)).exists()
+        if not track:
+            return {'error': 'track_wrong'}
+        try:
+            track.check_access_rights('read')
+            track.check_access_rule('read')
+        except AccessError:
+            return {'error': 'track_access'}
+        return {'track': track}
+
+    def _event_track_get_values(self, event, track, **options):
+        values = super(WebsiteEventTrackQuiz, self)._event_track_get_values(event, track)
+        if 'quiz' in options:
+            values.update({
+                'show_quiz': True,
+                'visitor': self._get_event_track_visitor_info(track)
+            })
+        return values
+
+    def _get_event_track_visitor_info(self, track):
+        track_visitor = request.env['event.track.visitor'].sudo().search([('track_id', '=', track.id), ('partner_id', '=', request.env.user.partner_id.id)])
```

Comment:
I would do a '_find_track_visitor' on track(s), that returns track.visitors based on current user's partner (if not public) or current visitor (with self.env['website.visitor']._get_visitor_from_request(force_create=False)). Even if visitors are not supported yet, method will be complete :) .